xray/audit/jas/jasmanager.go

```diff
@@ -12,50 +13,93 @@ import (
 	"github.com/owenrumney/go-sarif/v2/sarif"
 	"gopkg.in/yaml.v3"
 	"os"
+	"path/filepath"
 )
 
 var (
-	analyzerManagerExecuter utils.AnalyzerManagerInterface = &utils.AnalyzerManager{}
-	skippedDirs                                            = []string{"**/*test*/**", "**/*venv*/**", "**/*node_modules*/**", "**/*target*/**"}
+	skippedDirs = []string{"**/*test*/**", "**/*venv*/**", "**/*node_modules*/**", "**/*target*/**"}
 )
 
+type ScannerCmd interface {
+	Run(wd string) (err error)
+}
+
+type AdvancedSecurityScanner struct {
+	configFileName        string
+	resultsFileName       string
+	analyzerManager       utils.AnalyzerManager
+	serverDetails         *config.ServerDetails
+	workingDirs           []string
+	scannerDirCleanupFunc func() error
+}
+
+func NewAdvancedSecurityScanner(workingDirs []string, serverDetails *config.ServerDetails) (scanner *AdvancedSecurityScanner, err error) {
+	scanner = &AdvancedSecurityScanner{}
+	if scanner.analyzerManager.AnalyzerManagerFullPath, err = utils.GetAnalyzerManagerExecutable(); err != nil {
+		return
+	}
+	var tempDir string
+	if tempDir, err = fileutils.CreateTempDir(); err != nil {
+		return
+	}
+	scanner.scannerDirCleanupFunc = func() error {
+		return fileutils.RemoveTempDir(tempDir)
+	}
+	scanner.serverDetails = serverDetails
+	scanner.configFileName = filepath.Join(tempDir, "config.yaml")
+	scanner.resultsFileName = filepath.Join(tempDir, "results.sarif")
+	scanner.workingDirs, err = utils.GetFullPathsWorkingDirs(workingDirs)
+	return
+}
+
+func (a *AdvancedSecurityScanner) Run(scannerCmd ScannerCmd) (err error) {
+	for _, workingDir := range a.workingDirs {
+		func() {
+			defer func() {
+				err = errors.Join(err, deleteJasProcessFiles(a.configFileName, a.resultsFileName))
```

Comment:
why not a.deleteJasProcessFiles() ?

Reply:
@sverdlov93, there is no need for any of the struct parameters, so I don't think adding this function as a method will make it clearer/better.